GoogleDataTransport/GDTCORLibrary/Public/GDTCORStorageProtocol.h

```diff
@@ -36,6 +38,31 @@ NS_ASSUME_NONNULL_BEGIN
 /** Removes the events from storage. */
 - (void)removeEvents:(NSSet<NSNumber *> *)eventIDs;
 
+/** Persists the given data with the given key.
+ *
+ * @param data The data to store.
+ * @param key The unique key to store it to.
+ * @param onComplete An block to be run when storage of the data is complete.
+ */
+- (void)storeLibraryData:(NSData *)data
+                  forKey:(NSString *)key
+              onComplete:(void (^)(NSError *_Nullable error))onComplete;
+
+/** Retrieves the stored data for the given key.
+ *
+ * @param key The key corresponding to the desired data.
+ * @param onComplete The callback to invoke with the data once it's retrieved.
+ */
+- (void)libraryDataForKey:(NSString *)key onComplete:(void (^)(NSData *_Nullable data))onComplete;
```

Comment:
Are there any potential read errors we may want to expose in the completion?

Reply:
Done.